service/storage_service.cc

```diff
@@ -2957,21 +2957,23 @@ future<> storage_service::removenode_with_stream(gms::inet_address leaving_node,
 }
 
 future<> storage_service::restore_replica_count(inet_address endpoint, inet_address notify_endpoint) {
+    auto sas = make_shared<abort_source>();
+    _abort_source.check();
```

Comment:
Maybe do this before allocating?

Reply:
Won't `subscribe()` below throw anyway BTW?

Reply:
> Maybe do this before allocating?

Yeah, that can be done

Reply:
Fixed in v2